webapp/src/components/rhs_sidebar/rhs_sidebar.tsx

```diff
@@ -125,34 +124,37 @@ const Subscriptions = (props) => {
     );
 };
 
-Subscriptions.propTypes = {
-    currentChannelId: PropTypes.string,
-    subscriptions: PropTypes.arrayOf(PropTypes.shape({
-        repository_name: PropTypes.string.isRequired,
-        repository_url: PropTypes.string.isRequired,
-    })).isRequired,
-};
+interface SubscriptionsProps {
+    currentChannelId: string;
+    subscriptions: Subscription[];
+}
 
-export default class RHSSidebar extends React.PureComponent {
-    static propTypes = {
-        currentUserId: PropTypes.string.isRequired,
-        connected: PropTypes.bool.isRequired,
-        username: PropTypes.string,
-        gitlabURL: PropTypes.string,
-        currentChannelId: PropTypes.string,
-        currentChannelSubscriptions: PropTypes.arrayOf(PropTypes.shape({
-            repository_name: PropTypes.string,
-            repository_url: PropTypes.string,
-            features: PropTypes.arrayOf(PropTypes.string),
-        })).isRequired,
-        pluginServerRoute: PropTypes.string.isRequired,
-        actions: PropTypes.shape({
-            getChannelSubscriptions: PropTypes.func.isRequired,
-            sendEphemeralPost: PropTypes.func.isRequired,
-        }).isRequired,
-    };
+interface Subscription {
+    repository_name: string;
+    repository_url: string;
+    features: string[];
+}
+
+interface RhsSidebarProps {
+    currentUserId: string,
+    connected: boolean,
+    username: string,
+    gitlabURL: string,
+    currentChannelId: string,
```

Comment:
These fields were not required earlier and I think they can be undefined. What are your opinions on this?